packages/domain/lib/mappers/people_mapper.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:domain/mappers/base_mappers.dart';
+import 'package:domain/model/cast.dart';
+import 'package:domain/model/people_and_images_model.dart';
+import 'package:domain/model/tmdb_response.dart';
+import 'package:domain/utils/const.dart';
+
+class PeopleMapper extends MapperTwoParams<List<Cast>,
+    List<TMDBResponse>, List<PeopleAndImagesModel>> {
+  @override
+  List<PeopleAndImagesModel> call(
```

Comment:
need to think over the mapping logic
now u have the following flow

1) receive the list
2) for each item u call the api to get the image ( O(n) )
3) add response from image to list with images responses
4) map 2 lists ( O(n^2) - because while mapping u iterate the list on each element)

but at the same time, if u will map the image response with cast right after u received image response, your complexity will be O(n)

Reply:
changed